connection.go

```diff
@@ -640,9 +640,12 @@ func (c *Connection) dispatch0(f frame) {
 		// kthx - all reads reset our deadline.  so we can drop this
 	default:
 		// lolwat - channel0 only responds to methods and heartbeats
-		if err := c.closeWith(ErrUnexpectedFrame); err != nil {
-			Logger.Printf("error sending connectionCloseOk with ErrUnexpectedFrame, error: %+v", err)
-		}
+		// closeWith use call don't block reader
+		go func() {
+			if err := c.closeWith(ErrUnexpectedFrame); err != nil {
+				Logger.Printf("error sending connectionCloseOk with ErrUnexpectedFrame, error: %+v", err)
+			}
+		}()
```

Comment:
This approach has a lot of potential to cause routine leaks. What's the difference between using a go routine and simply synchronously call the function?

See the following test failure what detected a go routine leak: https://github.com/rabbitmq/amqp091-go/actions/runs/6324272238/job/17185854481?pr=223#step:5:333

Reply:
https://github.com/rabbitmq/amqp091-go/issues/222#issuecomment-1736798083

> n some case, reader will call c.closeWith, and block at c.call at <- c.rpc. But only reader write to <-c.rpc - -

`dispatch0` and `dispatchN` is uesed at `reader` function. If we sync call this fucniton, `reader` will block.

Reply:
fixed

Reply:
I'm happy with the changes in f4ea4b2b288a3863365f021f45a429099e305086